packages/devtools_app/lib/src/debugger/flutter/call_stack.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart' hide Stack;
+import 'package:provider/provider.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../flutter/theme.dart';
+import 'debugger_controller.dart';
+import 'debugger_model.dart';
+
+class CallStack extends StatefulWidget {
+  const CallStack({Key key}) : super(key: key);
+
+  @override
+  _CallStackState createState() => _CallStackState();
+}
+
+class _CallStackState extends State<CallStack> {
+  DebuggerController controller;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    final newController = Provider.of<DebuggerController>(context);
+    if (newController == controller) return;
+    controller = newController;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<List<StackFrameAndSourcePosition>>(
+      valueListenable: controller.stackFramesWithLocation,
+      builder: (context, stackFrames, _) {
+        return ValueListenableBuilder<StackFrameAndSourcePosition>(
+          valueListenable: controller.selectedStackFrame,
+          builder: (context, selectedFrame, _) {
+            return ListView.builder(
+              itemCount: stackFrames.length,
+              itemExtent: defaultListItemHeight,
+              itemBuilder: (_, index) {
+                final frame = stackFrames[index];
+                final selected =
```

Comment:
You may want to just have `final selected = frame == selectedFrame`, and assume that code outside of here will select the first frame when it should.